Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.UnitTests/Services/ApiGatewayRouteConfigServiceTests.cs

```diff
@@ -0,0 +1,256 @@
+    {
+        // Arrange
+        var routeConfig = new ApiGatewayRouteConfig
+        {
+            LambdaResourceName = "TestLambdaFunction",
+            HttpMethod = "POST",
+            Path = "/test/{id}"
+        };
+
+        _mockEnvironmentManager
+            .Setup(m => m.GetEnvironmentVariables())
+            .Returns(new Dictionary<string, string>
+            {
+                { Constants.LambdaConfigEnvironmentVariablePrefix, JsonSerializer.Serialize(routeConfig) }
+            });
+
+        var service = new ApiGatewayRouteConfigService(_mockEnvironmentManager.Object, _mockLogger.Object);
+
+        // Act
+        var result = service.GetRouteConfig("GET", "/test/123");
+
+        // Assert
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public void GetRouteConfig_ReturnsNullForNonMatchingPath()
+    {
+        // Arrange
+        var routeConfig = new ApiGatewayRouteConfig
+        {
+            LambdaResourceName = "TestLambdaFunction",
+            HttpMethod = "GET",
+            Path = "/test/{id}"
+        };
+
+        _mockEnvironmentManager
+            .Setup(m => m.GetEnvironmentVariables())
+            .Returns(new Dictionary<string, string>
+            {
+                { Constants.LambdaConfigEnvironmentVariablePrefix, JsonSerializer.Serialize(routeConfig) }
+            });
+
+        var service = new ApiGatewayRouteConfigService(_mockEnvironmentManager.Object, _mockLogger.Object);
+
+        // Act
+        var result = service.GetRouteConfig("GET", "/nonexistent/123");
+
+        // Assert
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public void Constructor_LoadsAndParsesListOfConfigs()
+    {
+        // Arrange
+        var routeConfigs = new List<ApiGatewayRouteConfig>
+        {
+            new ApiGatewayRouteConfig
+            {
+                LambdaResourceName = "Function1",
+                HttpMethod = "GET",
+                Path = "/path1"
+            },
+            new ApiGatewayRouteConfig
+            {
+                LambdaResourceName = "Function2",
+                HttpMethod = "POST",
+                Path = "/path2"
+            }
+        };
+
+        _mockEnvironmentManager
+            .Setup(m => m.GetEnvironmentVariables())
+            .Returns(new Dictionary<string, string>
+            {
+                { Constants.LambdaConfigEnvironmentVariablePrefix, JsonSerializer.Serialize(routeConfigs) }
+            });
+
+        var service = new ApiGatewayRouteConfigService(_mockEnvironmentManager.Object, _mockLogger.Object);
+
+        // Act
+        var result1 = service.GetRouteConfig("GET", "/path1");
+        var result2 = service.GetRouteConfig("POST", "/path2");
+
+        // Assert
+        Assert.NotNull(result1);
+        Assert.NotNull(result2);
+        Assert.Equal("Function1", result1.LambdaResourceName);
+        Assert.Equal("Function2", result2.LambdaResourceName);
+    }
+
+    [Fact]
+    public void ProperlySortRouteConfigs()
```

Comment:
Do you verify with an API Gateway deployed with routes like this that returns behaves the same?

Reply:
Yes the values I have in the test have been verified in API Gateway